src/user/dto/create-user.dto.ts

```diff
@@ -1 +1,48 @@
-export class CreateUserDto {}
+import {
+  IsAlphanumeric,
+  IsEmail,
+  IsEnum,
+  IsInt,
+  IsNotEmpty,
+  IsString,
+  Matches,
+  MinLength,
+} from 'class-validator';
+
+const passwordRegEx =
+  /^(?=.*[a-z])(?=.*[A-Z])(?=.*d)(?=.*[@$!%*?&])[A-Za-zd@$!%*?&]{8,20}$/;
+
+export class CreateUserDto {
+  @IsString()
+  @MinLength(2, { message: 'Name must have atleast 2 characters.' })
+  @IsNotEmpty()
+  name: string;
+
+  @IsNotEmpty()
+  @MinLength(3, { message: 'Username must have atleast 3 characters.' })
+  @IsAlphanumeric(null, {
+    message: 'Username does not allow other than alpha numeric chars.',
+  })
+  username: string;
+
+  @IsNotEmpty()
+  @IsEmail(null, { message: 'Please provide valid Email.' })
+  email: string;
+
+  @IsInt()
+  age: number;
+
+  @IsString()
+  @IsEnum(['f', 'm', 'u'])
```

Comment:
better to use real Enum instead of string array, that also will be displayed in your database like enum. e.g. enum Gender {...}

Reply:
nice comment